src/main/java/net/syscon/prison/repository/jpa/model/ExternalMovement.java

```diff
@@ -0,0 +1,110 @@
+package net.syscon.prison.repository.jpa.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.hibernate.annotations.JoinColumnOrFormula;
+import org.hibernate.annotations.JoinColumnsOrFormulas;
+import org.hibernate.annotations.JoinFormula;
+import org.hibernate.annotations.NotFound;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.Id;
+import javax.persistence.IdClass;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import java.io.Serializable;
+import java.time.LocalDateTime;
+
+import static javax.persistence.EnumType.STRING;
+import static net.syscon.prison.repository.jpa.model.City.CITY;
+import static net.syscon.prison.repository.jpa.model.MovementReason.REASON;
+import static net.syscon.prison.repository.jpa.model.MovementType.TYPE;
+import static org.hibernate.annotations.NotFoundAction.IGNORE;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name = "OFFENDER_EXTERNAL_MOVEMENTS")
+@IdClass(ExternalMovement.PK.class)
+public class ExternalMovement extends AuditableEntity {
+
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class PK implements Serializable {
+        @Column(name = "OFFENDER_BOOK_ID", updatable = false, insertable = false)
```

Comment:
You can map to complex types for keys as well if you want...

Reply:
as discussed, that doesn't seem to work in this case.